public/app/plugins/datasource/cloudwatch/datasource.ts

```diff
@@ -69,7 +69,7 @@ export default class CloudWatchDatasource extends DataSourceApi<CloudWatchQuery,
       return (
         (item.id !== '' || item.hide !== true) &&
         ((!!item.region && !!item.namespace && !!item.metricName && !_.isEmpty(item.statistics)) ||
-          item.expression.length > 0)
+          (item.expression && item.expression.length > 0))
```

Comment:
Would this improve readability?
```suggestion
          item.expression?.length > 0)
```

Reply:
Thanks, looks better!